src/engine/Operation.h

```diff
@@ -34,20 +34,37 @@ class Operation {
   // Use existing results if they are already available, otherwise
   // trigger computation.
   shared_ptr<const ResultTable> getResult() const {
-    LOG(DEBUG) << "Try to atomically emplace a new empty ResultTable" << endl;
+    LOG(DEBUG) << "Check cache for Operation result" << endl;
     LOG(DEBUG) << "Using key: \n" << asString() << endl;
     auto [newResult, existingResult] =
         _executionContext->getQueryTreeCache().tryEmplace(asString());
     if (newResult) {
-      LOG(DEBUG) << "We were the first to emplace, need to compute result"
-                 << endl;
+      LOG(DEBUG) << "Not in the cache, need to compute result" << endl;
       // Passing the raw pointer here is ok as the result shared_ptr remains
       // in scope
-      computeResult(newResult.get());
+      try {
+        computeResult(newResult.get());
+      } catch (const ad_semsearch::Exception& e) {
```

Comment:
ad:semsearch::Exception should inherit from std::exception and implement what().
ad_semsearch::AbortException should be a separate type.
Then you can write one catch block for AbortException (simply rethrow) and one for all other exceptions (print the `what()` message and throw and AbortException. In particular we also want to handle `std::bad_alloc` etc.